nexus-writer/src/nexus/engine.rs

```diff
@@ -170,6 +170,14 @@ impl NexusEngine {
                 if let Err(e) = run.end_span() {
                     warn!("Run span drop failed {e}")
                 }
+                if let Some((file_name, archive_name)) = Option::zip(
+                    self.filename.as_ref(),
+                    self.nexus_settings.archive_path.as_ref(),
+                ) {
+                    if let Err(e) = run.move_to_archive(file_name, archive_name) {
```

Comment:
You are not `await`ing the `JoinHandle`, so there is no guarantee the move ever happens.

Reply:
Wouldn't that only be a problem if the program exits unexpectedly? As it is running continuously the move should always happen in the background, shouldn't it?

Reply:
Realistically, yes, theoretically, no.
Plus it failing just because the writer exits is not a good design (keep in mind that the NeXus writer might be recycled between runs if the pipeline configuration changes).